### PR TITLE
Feature/ikdt 607 membersOf on Searcher class 

### PR DESCRIPTION
**Pull request by dettmoney**
Adding new method to Searcher class to get all concepts matching the provided pattern.

### CODE REVIEW

File: integration/src/test/java/dev/ikm/tinkar/integration/search/SearcherIT.java [hunk 1]
```diff
@@ -121,4 +126,26 @@ public void searchFromDescendantsOfConceptWithCustomCalculatorIT() throws Except
         // 2) Refrenced component type restriction
         assertEquals(6, searchResults.size(), "Exactly 6 search results should be returned");
     }
+
+    @Test
+    public void searchConceptsWithMembershipSemantic() throws Exception {
+        // test memberPatternId does not exist
+        EntityProxy.Concept conceptProxy = EntityProxy.Concept.make(PublicIds.newRandom());
+        List<PublicId> conceptIds = Searcher.membersOf(conceptProxy.publicId());
+        assertTrue(conceptIds.isEmpty(), "memberPatternId does not exist, should return empty list");
```
Comment (stenkarl) [inline integration/src/test/java/dev/ikm/tinkar/integration/search/SearcherIT.java:135]: Could each of these assertions be separate tests? I know these aren't unit tests, so they are more costly to run, so if it loads up up a DB for each test, probably best to leave them all together.

File: provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java [hunk 2]
```diff
@@ -344,4 +346,25 @@ public static List<PublicId> getAllowedResultsFromResultConformance(NavigationCa
...
+     * @return  List of PublicIds for the Concepts tagged with the Membership Pattern
+     */
+    public static List<PublicId> membersOf(PublicId memberPatternId) {
+        List<PublicId> conceptIds = new ArrayList<>();
+        try {
+            EntityService.get().getEntity(EntityService.get().nidForPublicId(memberPatternId)).ifPresent((e) -> {
+                if (e instanceof PatternEntity<?> patternEntity) {
+                    EntityService.get().forEachSemanticOfPattern(patternEntity.nid(), (semanticEntityOfPattern) -> {
+                        conceptIds.add(semanticEntityOfPattern.referencedComponent().publicId());
+                    });
```
Comment (stenkarl) [inline provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java:362]: IntelliJ tells me that you can simplify this expression to a lambda expression if you want to save a few characters:

```suggestion
EntityService.get().forEachSemanticOfPattern(patternEntity.nid(), (semanticEntityOfPattern) -> conceptIds.add(semanticEntityOfPattern.referencedComponent().publicId()));
```

Comment (dettmoney) [reply provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java:362]: Thanks - will fix this up.

File: integration/src/test/java/dev/ikm/tinkar/integration/search/SearcherIT.java (hunk 1 above)
Comment (rbradley813) [reply integration/src/test/java/dev/ikm/tinkar/integration/search/SearcherIT.java:135]: @stenkarl, great suggestion! @dettmoney the database is started in the BeforeAll method so we should be able to split these tests up without the performance hit of starting / stopping the database each time.

File: provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java [hunk 3]
```diff
@@ -344,4 +346,25 @@ public static List<PublicId> getAllowedResultsFromResultConformance(NavigationCa
...
+    /**
+     * Returns List of PublicIds for the Concepts tagged with the Membership Pattern
+     *
+     * @param   memberPatternId PublicId of the Membership Pattern
+     * @return  List of PublicIds for the Concepts tagged with the Membership Pattern
+     */
+    public static List<PublicId> membersOf(PublicId memberPatternId) {
+        List<PublicId> conceptIds = new ArrayList<>();
+        try {
+            EntityService.get().getEntity(EntityService.get().nidForPublicId(memberPatternId)).ifPresent((e) -> {
```
Comment (rbradley813) [inline provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java:358]: @dettmoney suggest using one of the other overloaded methods for getEntity to reduce the number of calls to EntityService.

**Review by stenkarl — APPROVED** [review]

**Pull request merged**